File: src/Shared/Polyrific.Catapult.Shared.ApiClient/ApplicationSettingService.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Polyrific.Catapult.Shared.Dto.ApplicationSetting;
using Polyrific.Catapult.Shared.Service;

namespace Polyrific.Catapult.Shared.ApiClient
{
    public class ApplicationSettingService : BaseService, IApplicationSettingService
    {
        public ApplicationSettingService(IApiClient api) : base(api)
        {
        }

        public async Task<List<ApplicationSettingDto>> GetApplicationSettings()
        {
            var path = $"applicationsetting";

            return await Api.Get<List<ApplicationSettingDto>>(path);
        }

        public async Task<ApplicationSettingValueDto> GetApplicationSettingValue()
        {
            var path = $"applicationsetting/value";

            return await Api.Get<ApplicationSettingValueDto>(path);
        }

        public async Task UpdateApplicationSetting(UpdateApplicationSettingDto dto)
        {
            var path = $"applicationsetting";

            await Api.Put(path, dto);
        }
    }
}

File: src/Web/Polyrific.Catapult.Web/ClientApp/src/app/core/services/application-setting.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { ApplicationSettingValueDto } from '../models/setting/application-setting-value-dto';
import { ApplicationSettingDto } from '../models/setting/application-setting-dto';
import { UpdateApplicationSettingDto } from '../models/setting/update-application-setting-dto';

@Injectable()
export class ApplicationSettingService {

  constructor(private api: ApiService) { }

  getApplicationSettings() {
    return this.api.get<ApplicationSettingDto[]>('applicationsetting');
  }

  getApplicationSettingValue() {
    return this.api.get<ApplicationSettingValueDto>('applicationsetting/value');
  }

  updateApplicationSetting(dto: UpdateApplicationSettingDto) {
    return this.api.put('applicationsetting', dto);
  }
}
